src/devices/bus/nes_ctrl/konamibag.cpp

```diff
@@ -0,0 +1,88 @@
+// license:BSD-3-Clause
+// copyright-holders:kmg, Fabio Priuli
+/**********************************************************************
+
+    Nintendo Family Computer - Konami Exciting Boxing Air Bag
+
+**********************************************************************/
+
+#include "emu.h"
+#include "konamibag.h"
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(NES_KONAMIBAG, nes_konamibag_device, "nes_konamibag", "Konami Exciting Boxing Air Bag")
+
+
+static INPUT_PORTS_START( nes_konamibag )
+	PORT_START("BAG.0")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_CODE(KEYCODE_7_PAD) PORT_NAME("Left Hook")
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_CODE(KEYCODE_3_PAD) PORT_NAME("Move Right")
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_CODE(KEYCODE_1_PAD) PORT_NAME("Move Left")
```

Comment:
Would it make sense to make these two left/right rather than buttons?  Is it possible to activate them both at the same time?  Are they in the mat you stand on, or do they respond to punches as well?  So many questions.

Reply:
Sorry, I have the same questions. I tried looking for clues in pictures of the mat. I tried looking at some Russian guy's video of him playing with the real inflatable. I'm assuming they are power-pad-like buttons in the mat but????? It's pretty clear the documented known knowns are just deduced from the I/O read/writes.

Reply:
Here's a long Russian video: https://www.youtube.com/watch?v=CW1GgfdpPI0. Anybody understand Russian? Does he mention the left/right movement ever? The controller itself is rare and expensive so this may be the only hands-on, recording of it out there.

Reply:
> maybe he describes the movement part in Russian?

he says player can not control movement. there is 11 sensors in total located at bag body - 5 at left side, 5 at right side, 1 at very top. that's all, no any sensors at bottom part or floor mat.
lmk if you have more questions.

Reply:
Thank you, p1pkin.

Reply:
I see. What is described on NesDev as move left/right is actually the opponent boxer dodging left or right or something to that effect, now that I look at their animation with new eyes. I wonder how that is triggered exactly...I believe I can hear the accompanying sound effect in the video.

Reply:
@0kmg no problem, welcome.
but now I'm wondering why this PR adds 8 input lines while this device should have 11 ?
also, I think the proper way to emulate this thing and the way it played - use clickable artwork.

Reply:
@p1pkin Thanks for the additional feedback. Unfortunately, I don't know if there's enough info at this time to emulate this better, much less with clickable artwork. Are there really exactly 11 sensors involved, or is that just what the Youtuber believes from physically inspecting the exterior of the bag? Could there be even more? However many sensors and whatever their type how does the data get combined by the air bag's logic into the two nibbles the Famicom reads? I can't even really speculate at this point. I'll probably have to set this aside on the indefinite back burner.